adr/ADR-42.md

```diff
@@ -223,6 +224,31 @@ type JSConsumerGroupUnPinnedAdvisory struct {
 }
 ```
 
+### `priority_client` policy
+
+Users want to deliver to pull requests to clients not in a round-robin fashion but sorted by priority.
+
+Imagine resources in `us-east` are fetching work off a Stream but `us-east` is resource constrained. For reasons of latency and cross region bandwidth costs the desire is that whenever there are available resources in `us-east` they should first get the work else other regions like `us-west` and `eu-west` should receive the work.
```

Comment:
I think we should be more specific here or give an example:

`priority 3` will pick up work only if there are no pending pulls with `priority < 3`